From: Hannes Reinecke <hare@suse.de>
Date: Thu, 9 Jun 2011 12:15:57 +0200
Subject: [PATCH] cgroups: Add WARN_ON on empty subsystems
References: FATE#311054
Patch-Mainline: n/a

Normally the subsystem array shouldn't be empty.
However, this doesn't seem to be the case here.
So add a WARN_ON statement to aid debugging.

Signed-off-by: Hannes Reinecke <hare@suse.de>
---
 kernel/cgroup.c |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/kernel/cgroup.c b/kernel/cgroup.c
index 3cf6e88..f16de68 100644
--- a/kernel/cgroup.c
+++ b/kernel/cgroup.c
@@ -3848,7 +3848,8 @@ void cgroup_exit(struct task_struct *tsk, int run_callbacks)
 		 */
 		for (i = 0; i < CGROUP_SUBSYS_COUNT; i++) {
 			struct cgroup_subsys *ss = subsys[i];
-			if (ss->exit) {
+			WARN_ON(!ss);
+			if (ss && ss->exit) {
 				struct cgroup *old_cgrp =
 					rcu_dereference(cg->subsys[i])->cgroup;
 				struct cgroup *cgrp = task_cgroup(tsk, i);
-- 
1.6.0.2

